spec/solver_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rspec'
+require_relative '../solver'
+
+RSpec.describe Solver do
+  describe '#factorial' do
+    subject { Solver.new }
+    context 'check the factorial of 4 is 24 or not' do
+      it 'check the factorial' do
+        fac = subject.factorial(4)
+        expect(fac).to eql(24)
+      end
+    end
+
+    context 'Check the factorial of -12 is Argument Error' do
+      it 'Check the factorial by negative num' do
+        expect { subject.factorial(-2) }.to raise_error('Negative numbers not allowed')
+      end
+    end
+    context 'Check add using in correct output' do
+      it 'Check the false value of factorial 10' do
+        fac = subject.factorial(10)
+        expect(fac).not_to eql(100)
+      end
```

Comment:
Great job @ZuhRa-HashiMi 🤝

- [ ] As stated in the project requirements, kindly add the test for special-case - when input is 0, the result should equal 1.
- [ ] Kindly ensure your commit history shows clearly that you started with writing tests before writing the code. However, according to your commit history, it actually shows that you wrote the code before writing tests.
![tests](https://user-images.githubusercontent.com/63917725/221816731-2d399739-97f6-4247-b37e-7f8eaf38a437.PNG)